### ISSUE TITLE
Can't connect to Condens 9000i via MBLani 

### ISSUE THREAD

**Issue by Manuthemanu** [bosch-thermostat/home-assistant-bosch-custom-component#209]
**Describe the bug**

I found out the IP-Adress of my Bosch Gastherme Condens 9000i which is connect via MBLani Ver. 02. When tying to integrate in Bosch Thermostat by IVT, HTTP, it failed by: Falsche Anmeldedaten oder falsche IP (Wrong credentials or wrong IP)

So I tried a scan for Bosch devices my network.

```
bosch_cli scan --host 074823048 --token As8eGS6n3Rr41t5R --password ja39.djb3m --protocol HTTP --device IVT`
Numbers are just an example
```

I get the following message:

```
2022-10-06 16:46:15 ERROR (MainThread) [bosch_thermostat_client.gateway.ivt] I cannot find supported device. Your devices: null
Traceback (most recent call last):
  File "../HomeAssistant/bosch/venv/bin/bosch_cli", line 8, in <module>
    sys.exit(cli())
  File "../HomeAssistant/bosch/venv/lib/python3.8/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "../HomeAssistant/bosch/venv/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "../HomeAssistant/bosch/venv/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "../HomeAssistant/bosch/venv/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "../HomeAssistant/bosch/venv/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "../HomeAssistant/bosch/venv/lib/python3.8/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "../HomeAssistant/bosch/venv/lib/python3.8/site-packages/bosch_thermostat_client/bosch_cli.py", line 86, in wrapper
    return asyncio.run(f(*args, **kwargs))
  File "/usr/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "../HomeAssistant/bosch/venv/lib/python3.8/site-packages/bosch_thermostat_client/bosch_cli.py", line 241, in scan
    connected = True if ignore_unknown else await gateway.check_connection()
  File "../HomeAssistant/bosch/venv/lib/python3.8/site-packages/bosch_thermostat_client/gateway/base.py", line 323, in check_connection
    await self.initialize()
  File "../HomeAssistant/bosch/venv/lib/python3.8/site-packages/bosch_thermostat_client/gateway/base.py", line 92, in initialize
    raise UnknownDevice("Your device is unknown %s" % json.dumps(self._device))
bosch_thermostat_client.exceptions.UnknownDevice: Your device is unknown null
```

Does anyone has a glue what to do next?
Thanks for help

**Comment by honikos** [comment]
This guy did it (use google translator)
https://smarterkram.de/2126/#integration-gasheizung-bosch-condens-9000i-wm-in-home-assistant
maybe it helps

**Comment by Manuthemanu** [comment]
> This guy did it (use google translator) https://smarterkram.de/2126/#integration-gasheizung-bosch-condens-9000i-wm-in-home-assistant maybe it helps

Thanks, I actually used this instruction, but it will not work.

**Comment by honikos** [comment]
Do you have special characters in your password? Try one only with letters and numbers. Without blank space, ... I read somewhere about it when I connected my device with home-assistant bosch custom component. I changed my password.

**Comment by Manuthemanu** [comment]
Yes I do. But it's not possible to change it. If I change it, it says: use at least:
- 10 charaters
- 1 special charater
- 1 Number
- 1 big Letter

**Comment by honikos** [comment]
ok, with my gateway or with my app it's not necessary to follow these password rules.
So this could be the problem.
BTW: in your scan you used a password without 1 big letter ( ja39.djb3m ). Is this the problem? (Numbers are just an example?)

**Comment by olli-dot-dev** [comment]
Here's the "guy" from smarterkram.de ;)

What app are you using for changing the password? There are many bosch apps out there unfortuanetly! Back then I managed to change my password using the Bosch EasyControl App. As of now, I'm having the HomeCom Easy app running on a virtual machine though, just to check that everything is fine, once in a while.

I remember that using the "right" bosch app was mandatory in order to get everything running!

**Comment by Manuthemanu** [comment]
Hi, how nice you wrote me here. Your hint was the right one. Actually, I use the HomeCOM App.
I tried the Easy Control App, but it says my device is not available. Finally, I used the IVT Anywhere App it allowed to integrate my device and I could set a password without special characters (If that is important or not). Instead of HTTP I used IVT, XAMPP and finally it worked. Then I followed the rest of your instruction and it looks like it's working now :) after many hours of trying. Now I just have to check if the data is right. Thanks for your help @honikos and @olli-dot-dev
